build/Wikibase/entrypoint.sh

```diff
@@ -1,59 +1,90 @@
+        if [ -z "${!var}" ]; then
+            echo "$var is required but isn't set. You should pass it to Docker. See: https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file"
+            exit 1
+        fi
+    done
+    set -u
 
-# Do the mediawiki install (only if LocalSettings doesn't already exist)
-if [ ! -e "/var/www/html/LocalSettings.php" ]; then
-    php /var/www/html/maintenance/install.php --dbuser "$DB_USER" --dbpass "$DB_PASS" --dbname "$DB_NAME" --dbserver "$DB_SERVER" --lang "$MW_SITE_LANG" --pass "$MW_ADMIN_PASS" "$MW_SITE_NAME" "$MW_ADMIN_NAME"
-    php /var/www/html/maintenance/resetUserEmail.php --no-reset-password "$MW_ADMIN_NAME" "$MW_ADMIN_EMAIL"
+    # Run MediaWiki install script, and update values
+    php /var/www/html/maintenance/run.php install \
+        --server "$MW_WG_SERVER" \
+        --scriptpath "/w" \
+        --dbuser "$DB_USER" \
+        --dbpass "$DB_PASS" \
+        --dbname "$DB_NAME" \
+        --dbserver "$DB_SERVER" \
+        --pass "$MW_ADMIN_PASS" \
+        --lang "$MW_WG_LANGUAGE_CODE" \
+        "$MW_WG_SITENAME" \
+        "$MW_ADMIN_NAME"
 
-    # Copy our LocalSettings into place after install from the template
-    # https://stackoverflow.com/a/24964089/4746236
-    export DOLLAR='$'
-    envsubst < /LocalSettings.php.template > /var/www/html/LocalSettings.php
+    # Include WBS customizations to generated LocalSettings.php
+    {
+        set +u
+        if [ -n "$ELASTICSEARCH_HOST" ]; then
+            echo "\$elasticsearchHost = '$ELASTICSEARCH_HOST';"
+        fi
+        set -u
+        echo
+        echo '# Insert any custom settings which should be ran BEFORE extensions here'
+        echo
+        echo 'include "/var/www/html/LocalSettings.wbs-extensions.php";'
+        echo
+        echo '# Insert any custom settings which should be ran AFTER extensions here'
+    } >> /var/www/html/LocalSettings.php
 
-    # Run update.php to install Wikibase
-    php /var/www/html/maintenance/update.php --quick
+    # Replace /config/LocalSettings.php with newly generated LocalSettings.php
+    cp /var/www/html/LocalSettings.php /config/LocalSettings.php
+    # Update the MW Admin email address (if this admin user doesn't already exist, a new one will be created)
+    php /var/www/html/maintenance/run.php resetUserEmail --no-reset-password "$MW_ADMIN_NAME" "$MW_ADMIN_EMAIL"
+
+    php /var/www/html/maintenance/run.php update --quick
 
-    # Run extrascripts on first run
     if [ -f /default-extra-install.sh ]; then
         # shellcheck disable=SC1091
-        source /default-extra-install.sh
+        bash /default-extra-install.sh
     fi
 
-    # Run extrascripts on first run
     if [ -f /extra-install.sh ]; then
         # shellcheck disable=SC1091
-        source /extra-install.sh
+        bash /extra-install.sh
     fi
 fi
 
-# Copy LocalSettings.php to a shared directory, if the image is being used with this shared directory existing
```

Comment:
This whole way of doing things is now replaced by having a required `config` volume.